### PR TITLE
compare upgradeconfig with upgradepolicy right before commence

### PR DESCRIPTION
**Pull request by bmeng**
### What type of PR is this?
feature

### Which Jira/Github issue(s) this PR fixes?
Fixes [OSD-5006](https://issues.redhat.com/browse/OSD-5006)

### Pre-checks (if applicable):
- [x] Tested latest changes against a cluster
- [x] Ran `make generate` command locally to validate code changes
- [ ] Included documentation changes with PR

### CODE REVIEW

File: pkg/controller/upgradeconfig/upgradeconfig_controller.go [hunk 1]
```diff
@@ -188,6 +189,29 @@ func (r *ReconcileUpgradeConfig) Reconcile(request reconcile.Request) (reconcile
 			return reconcile.Result{}, err
 		}
 
+		upCoMan, err := upgradeconfigmanager.NewBuilder().NewManager(r.client)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		remoteSpecs, err := upCoMan.GetRemoteSpec()
```
Comment (mrbarge) [inline pkg/controller/upgradeconfig/upgradeconfig_controller.go:196]: Why not use the existing `Refresh()` here? If the current policy is different from the cluster's current UpgradeConfig, the local UpgradeConfig still should to be updated to whatever the new policy is.

`Refresh()` will do that already, and returns a boolean value to indicate if there was indeed a difference in policy vs UpgradeConfig, which also negates the need for the logic further below.

Comment (mrbarge) [reply pkg/controller/upgradeconfig/upgradeconfig_controller.go:196]: Another note - am thinking this should be going in within the `if schedulerResult.IsReady` block, ie. it should only be executed when we've reached the time to upgrade. Where it's placed right now will result in a call/refresh to OCM on each reconcile whilst the UC is in a pending state.

**Comment by dofinn** [comment]
LGTM. Will let @mrbarge have final say :+1:

**Comment by mrbarge** [comment]
Gave this a test today. Scenario was:

- Set up a policy
- Waited for cluster to sync policy
- Deleted policy
- Created new policy with a different time/version 2 hours in the future
- Waited for cluster to reach 'upgrade' time of first policy
- Cluster replaced the policy with the new policy
- Cluster reported its next upgrade as ~2 hours in the future as expected.

This seems good to go. I will lgtm as soon as the tests pass.

**Pull request merged**